test/types.ts

```diff
@@ -0,0 +1,22 @@
+import { IncomingMessage, ServerResponse, createServer } from "node:http";
+import { expectTypeOf } from "expect-type";
+import onFinished, { isFinished } from "..";
+
+createServer((req, res) => {
+  onFinished(req, (err, req) => {
+    expectTypeOf(err).toEqualTypeOf<Error | null | undefined>();
+    expectTypeOf(req).toEqualTypeOf<IncomingMessage>();
+  });
+
+  onFinished(res, (err, res) => {
+    expectTypeOf(err).toEqualTypeOf<Error | null | undefined>();
+    expectTypeOf(res).toEqualTypeOf<
+      ServerResponse<IncomingMessage> & {
+        req: IncomingMessage;
+      }
+    >();
```

Comment:
```suggestion
    expectTypeOf(res).toEqualTypeOf<ServerResponse>();
```

I'm not sure why the just `ServerResponse` is not working.

Reply:
It could be because of some augmentation. You can try `.toExtend()` instead of `.toEqualTypeOf()`

Reply:
Thanks that works